### PR TITLE
Display error when image exceeds `ASSETS_TRANSFORM_IMAGE_MAX_DIMENSION` limit

### LINKED ISSUE #7996: Display image error "too large" or similar, if an image exceeds the ASSETS_TRANSFORM_IMAGE_MAX_DIMENSION env var limit
### Discussed in https://github.com/directus/directus/discussions/7983

<div type='discussions-op-text'>

<sup>Originally posted by **josias-r** September 12, 2021</sup>
If an image larger then the ASSETS_TRANSFORM_IMAGE_MAX_DIMENSION env var limit is uploaded, it is displayed the following way:

image

Instead of an errored out image, maybe inform the user why the image is not being displayed (since it's too large)</div>

### PR DESCRIPTION
**Pull request by azrikahar**
Fixes #7996

## Reported Bug

When an image exceeds the configured `ASSETS_TRANSFORM_IMAGE_MAX_DIMENSION` env variable limit, the image src will break. However we can still view it by clicking `zoom`:

qwNo9Fv0pa

## Solution

call the asset api and catch the error if the code is `ILLEGAL_ASSET_TRANSFORMATION`, then diplay an info icon with a message.

Update:

- moved error handling to `@error` event for 
- added "generic" error type to show image cannot be loaded, in case the error isn't `ILLEGAL_ASSET_TRANSFORMATION`. (not sure if this will ever happen?)
- added relevant translation keys.

## After Fix

yldCawqCIv

### Updated look:

- When error code is `ILLEGAL_ASSET_TRANSFORMATION`:

    chrome_kIh3yM9sk0

- When error code is generic:

    chrome_JlABTkzaME

## ~Uncertainty~

~calling the assets API and trigger it in the same try/catch might feel hacky.. but in a weird way it sort of makes sense? still likely that I'm reaching for a quick fix too soon, so do let me know if there's a more appropriate approach.~

~the error message is also not set in stone. I just referred it from the existing error message from the api:~

https://github.com/directus/directus/blob/9ba43c19f8c7ae4add0394ace7f90cbc524177e7/api/src/services/assets.ts#L95-L97

### CODE REVIEW

**Comment by benhaynes** [comment]
I defer to @rijkvanzanten on implementation, but for style:

Let's shorten this text and give it a max-width (eg: 300px) so it feels more contained. Potential text update below:

```
Image source too large to preview.
```

**Comment by josias-r** [comment]
Unfortunately, I‘m not at the computer to check this myself. But did you also think of the image in the list view of a collection? It was also erroring out

**Comment by azrikahar** [comment]
> Unfortunately, I‘m not at the computer to check this myself. But did you also think of the image in the list view of a collection? It was also erroring out

Ya I also noticed that as I was investigating this issue. but I felt like the image in list view + image in files preview are more similar in nature (based on their logic in the source code), so I'm planning to fix those in a separate PR. 👍

**Comment by azrikahar** [comment]
Update (see main post for screenshots):

- moved error handling to `@error` event for 
- added "generic" error type to show image cannot be loaded, in case the error isn't `ILLEGAL_ASSET_TRANSFORMATION`. (not sure if this will ever happen?)
- added relevant translation keys.

**Comment by rijkvanzanten** [comment]
> added "generic" error type to show image cannot be loaded, in case the error isn't ILLEGAL_ASSET_TRANSFORMATION. (not sure if this will ever happen?)

Smart! We don't expect it to, but anything can happen when it comes to computers and networks, so better safe than sorry 🙂

File: app/src/interfaces/file-image/file-image.vue [hunk 1]
```diff
@@ -7,7 +7,17 @@
 		</v-notice>
 
 		<div v-else-if="image" class="image-preview" :class="{ 'is-svg': image.type && image.type.includes('svg') }">
-			<img :src="src" alt="" role="presentation" />
+			<div v-if="imageError" class="image-error">
+				<v-icon large :name="imageError === 'illegalAssetTransformation' ? 'info_outline' : 'error_outline'" />
+				<span class="message">
+					{{
+						imageError === 'illegalAssetTransformation'
+							? t('image_errors.too_large_to_preview')
```
Comment (rijkvanzanten) [inline app/src/interfaces/file-image/file-image.vue:15]: We already have an "ERRORS" key under translations. Maybe we can simply use the "translateAPIError" utility for this purpose :thinking:

File: app/src/lang/translations/en-US.yaml [hunk 2]
```diff
@@ -561,6 +561,9 @@ upload_files_success: '{count} Files Uploaded'
 upload_pending: Upload Pending
 drag_file_here: Drag & Drop a File Here
 click_to_browse: Click to Browse
+image_errors:
+  generic: Cannot load image.
+  too_large_to_preview: Image source too large to preview.
```
Comment (rijkvanzanten) [inline app/src/lang/translations/en-US.yaml:566]: Lets move this to the more generic "errors" key

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**